acceptancetests/jujupy/client.py

```diff
@@ -2121,6 +2128,90 @@ def switch(self, model=None, controller=None):
             raise ValueError('No target to switch to has been given.')
         self.juju('switch', (':'.join(args),), include_e=False)
 
+    def scp(self, _from, _to):
+        self.juju('scp', (_from, _to))
+
+
+# caas `add-k8s` did not implement parsing kube config path via flag,
+# so parse it via env var ->  https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/loader.go#L44
+KUBE_CONFIG_PATH_ENV_VAR = 'KUBECONFIG'
+
+
+class CaasClient(ModelClient):
+
+    cloud_name = 'k8cloud'
+    kubectl_path = None
+    kube_home = None
+    kube_config_path = None
+
+    def _ensure_dir(self, path):
+        try:
+            os.makedirs(path)
+        except OSError:
+            if not os.path.isdir(path):
+                raise
+
+    def __prepare(self):
+        # to pick up the new tmp juju_home, have to do this!
+        self.kubectl_path = os.path.join(self.env.juju_home, 'kubectl')
+        self.kube_home = os.path.join(self.env.juju_home, '.kube')
+        self.kube_config_path = os.path.join(self.kube_home, 'config')
+
+    def cluster_up(self, bundle_path):
+        self.__prepare()
+
+        self.deploy_bundle(bundle_path, static_bundle=True)
+        # Wait for the deployment to finish.
+        self.wait_for_started()
+
+        # ensure kube home
+        self._ensure_dir(self.kube_home)
+        # ensure kube credentials
+        self.scp(_from='kubernetes-master/0:config', _to=self.kube_config_path)
+
+        # ensure kubectl by scp from master
+        self.scp(_from='kubernetes-master/0:/snap/bin/kubectl', _to=self.kubectl_path)
+
+        # ensure kube config env var
+        os.environ[KUBE_CONFIG_PATH_ENV_VAR] = self.kube_config_path
+        # ensure k8s cloud
+        self.add_caas_cloud()
+        log.debug('added caas cloud, now all clouds are -> \n%s', self.list_clouds())
+
+    def add_caas_cloud(self):
+        self.juju('add-k8s', self.cloud_name)
+
+    def add_caas_model(self, caas_model_name):
+        return self.add_model(env=self.env.clone(caas_model_name), cloud_region=self.cloud_name)
+
+    @property
+    def is_cluster_healthy(self):
+        try:
+            cluster_info = self._kubectl_cmd('cluster-info')
+            log.debug('cluster_info -> \n%s', cluster_info)
+            nodes_info = self._kubectl_cmd('get nodes')
+            log.debug('nodes_info -> \n%s', nodes_info)
+            return True
+        except RuntimeError:
+            return False
+
+    def _kubectl_cmd(self, args):
```

Comment:
kubectl is tied pretty close to caas? Should this be a 'private' method or perhaps just 'def kubectl(...)' much like juju(...) us for ModelClient.

This would expose the kubectl command (assuming it's needed) so a test author could do something like (assuming the suggestion above re: moving CaasClient within ModelClient):
```python

client.caas.kubectl('some', 'command')
```

Reply:
I was trying to make the cmd generic, so we will need to wrap related actions(like `is_cluster_healthy`) here rather than calling kubectl from assess script.